src/TemporaryFile.cpp

```diff
@@ -42,10 +43,23 @@ get_umask()
 } // namespace
 
 TemporaryFile::TemporaryFile(string_view path_prefix)
-  : path(std::string(path_prefix) + ".XXXXXX")
 {
-  Util::ensure_dir_exists(Util::dir_name(path));
+  Util::ensure_dir_exists(Util::dir_name(path_prefix));
+
+#ifdef _WIN32
```

Comment:
I think this special case warrants a comment explaining why `mkstemp()` isn't used.